src/main/java/com/fundynamic/d2tm/game/controls/Mouse.java

```diff
@@ -61,13 +72,36 @@ public void deselectEntity() {
                 ((Selectable) lastSelectedEntity).deselect();
             }
         }
+        if (selectedEntityBelongsToPlayerControlled()) {
+            movingCursor = false;
+        }
         lastSelectedEntity = null;
     }
 
+    private boolean selectedEntityBelongsToPlayerControlled() {
+        if (lastSelectedEntity == null) return false;
+        if (playerControlled == null) return false;
```

Comment:
Maybe I'm a little late to the party, but wouldn't it be better to use polymorphism for `Mouse` to simplify this kind of null-checking logic. I propose to implement 2 concrete mouse classes, e.g. `PlayerControlledMouse` and `HoverCellMouse`. Waddayathink?